### PR TITLE
remove public/private folders and add database for metadata

### PR DESCRIPTION
**Pull request by AbdelHedhili**
Signed-off-by: Abdelsalem <abdelsalem.hedhili@rte-france.com>

**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
feature

**What is the current behavior?** *(You can also link to an open issue here)*
public/private storage exist

**What is the new behavior (if this is a feature change)?**
public/private storage doesn't exist + added case metadata database

### CODE REVIEW

File: case-server/src/main/java/com/powsybl/caseserver/repository/CaseMetadataEntity.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+@Entity
+@Table(name = "caseMetadata")
+public class CaseMetadataEntity {
+
+    @Id
+    @Column(name = "id")
+    private UUID id;
+
+    @Column(name = "creationDate", nullable = false)
+    private LocalDateTime date;
```
Comment (SlimaneAmar) [inline case-server/src/main/java/com/powsybl/caseserver/repository/CaseMetadataEntity.java:35]: Rename by dateCreation so as not to confuse with the date of the case

File: case-server/src/main/java/com/powsybl/caseserver/repository/CaseMetadataEntity.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+@NoArgsConstructor
+@AllArgsConstructor
+@Getter
+@Setter
+@Entity
+@Table(name = "caseMetadata")
+public class CaseMetadataEntity {
+
+    @Id
+    @Column(name = "id")
```
Comment (SlimaneAmar) [inline case-server/src/main/java/com/powsybl/caseserver/repository/CaseMetadataEntity.java:31]: Autogenerate ?

File: case-server/src/main/java/com/powsybl/caseserver/repository/CaseMetadataEntity.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    private UUID id;
+
+    @Column(name = "creationDate", nullable = false)
+    private LocalDateTime date;
+
```
Comment (SlimaneAmar) [inline case-server/src/main/java/com/powsybl/caseserver/repository/CaseMetadataEntity.java:36]: The creation date must be moved in the directory server ?

**Review by SlimaneAmar — CHANGES_REQUESTED** [review]

File: case-server/src/main/java/com/powsybl/caseserver/repository/CaseMetadataEntity.java (hunk 1 above)
Comment (AbdelHedhili) [reply case-server/src/main/java/com/powsybl/caseserver/repository/CaseMetadataEntity.java:35]: field removed

File: case-server/src/main/java/com/powsybl/caseserver/repository/CaseMetadataEntity.java (hunk 3 above)
Comment (AbdelHedhili) [reply case-server/src/main/java/com/powsybl/caseserver/repository/CaseMetadataEntity.java:36]: field removed

File: case-server/src/main/java/com/powsybl/caseserver/repository/CaseMetadataEntity.java (hunk 2 above)
Comment (AbdelHedhili) [reply case-server/src/main/java/com/powsybl/caseserver/repository/CaseMetadataEntity.java:31]: I think it's not useful here. I need the id to create the repository in which the case will be stored. So it's better if I generate it myself as I need it before I create this entity.

File: case-server/src/main/java/com/powsybl/caseserver/repository/CaseMetadataEntity.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    @Column(name = "expirationDate", nullable = false)
```
[the remark this answers is not in the document]
Comment (AbdelHedhili) [reply case-server/src/main/java/com/powsybl/caseserver/repository/CaseMetadataEntity.java]: removed

**Review by SlimaneAmar — CHANGES_REQUESTED** [review]

File: case-server/src/main/java/com/powsybl/caseserver/ScheduledCaseCleaner.java [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+
+/**
+ * @author Abdelsalem Hedhili <abdelsalem.hedhili at rte-france.com>
+ */
+@Component
```
[the remark this answers is not in the document]
Comment (AbdelHedhili) [reply case-server/src/main/java/com/powsybl/caseserver/ScheduledCaseCleaner.java:23]: ok

**Review by jonenst — APPROVED** [review]

File: case-server/src/main/java/com/powsybl/caseserver/CaseService.java [hunk 6]
```diff
@@ -260,6 +245,14 @@ UUID createCase(UUID sourceCaseUuid) {
...
             CaseInfos caseInfos = createInfos(existingCaseInfos.getName(), newCaseUuid, existingCaseInfos.getFormat());
             caseInfosService.addCaseInfos(caseInfos);
+
+            LocalDateTime expirationTime = null;
+            if (withExpiration) {
+                expirationTime = LocalDateTime.now(ZoneOffset.UTC).plusHours(1);
+            }
+            CaseMetadataEntity caseEntity = new CaseMetadataEntity(newCaseUuid, expirationTime);
+            caseMetadataRepository.save(caseEntity);
+
```
[the remark this answers is not in the document]
Comment (AbdelHedhili) [reply case-server/src/main/java/com/powsybl/caseserver/CaseService.java:255]: done

File: case-server/src/main/java/com/powsybl/caseserver/ScheduledCaseCleaner.java [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+@ComponentScan(basePackageClasses = {CaseService.class, CaseMetadataRepository.class})
+public class ScheduledCaseCleaner {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ScheduledCaseCleaner.class);
+
+    @Autowired
+    private CaseMetadataRepository caseMetadataRepository;
```
[the remark this answers is not in the document]
Comment (AbdelHedhili) [reply case-server/src/main/java/com/powsybl/caseserver/ScheduledCaseCleaner.java:30]: done

File: case-server/src/main/java/com/powsybl/caseserver/CaseController.java [hunk 8]
```diff
@@ -156,12 +149,23 @@ public ResponseEntity<UUID> importPublicCase(@RequestParam("file") MultipartFile
...
-    public ResponseEntity<UUID> createCase(@RequestParam("duplicateFrom") UUID sourceCaseUuid) {
+    public ResponseEntity<UUID> createCase(@RequestParam("duplicateFrom") UUID sourceCaseUuid,
+                                           @RequestParam(value = "withExpiration", required = false, defaultValue = "false") boolean withExpiration) {
         LOGGER.info("createCase request received with parameter sourceCaseUuid = {}", sourceCaseUuid);
-        UUID newCaseUuid = caseService.createCase(sourceCaseUuid);
+        UUID newCaseUuid = caseService.createCase(sourceCaseUuid, withExpiration);
         return ResponseEntity.ok().body(newCaseUuid);
     }
 
+    @DeleteMapping(value = "/cases/{caseUuid}/expiration")
```
[the remark this answers is not in the document]
Comment (AbdelHedhili) [reply case-server/src/main/java/com/powsybl/caseserver/CaseController.java:159]: as discussed we're going with @PutMapping(value = "/cases/{caseUuid}/disableExpiration")

**Review by SlimaneAmar — APPROVED** [review]

**Pull request merged**